### PR TITLE
SCMOD-12164: Add message publish confirm and retry

### PR DESCRIPTION
**Pull request by dgibson-microfocus**
https://portal.digitalsafe.net/browse/SCMOD-12164

### CODE REVIEW

File: job-service/src/main/java/com/hpe/caf/services/job/queue/QueueServices.java [hunk 1]
```diff
@@ -124,8 +128,20 @@ public void sendMessage(
...
+            LOG.warn("Failed to publish message. Will retry", e);
+            publishMessage(taskMessageBytes);
+        }
+    }
+
+    private void publishMessage(final byte[] taskMessageBytes)
+            throws IOException, InterruptedException, TimeoutException
+    {
         publisherChannel.basicPublish(
                 "", targetQueue, MessageProperties.TEXT_PLAIN, taskMessageBytes);
```
Comment (dermot-hardy) [inline job-service/src/main/java/com/hpe/caf/services/job/queue/QueueServices.java:143]: Should this also be `MessageProperties.PERSISTENT_TEXT_PLAIN`?

**Review by dermot-hardy — APPROVED** [review]

File: job-service/src/main/java/com/hpe/caf/services/job/queue/QueueServices.java (hunk 1 above)
Comment (dgibson-microfocus) [reply job-service/src/main/java/com/hpe/caf/services/job/queue/QueueServices.java:143]: Yes probably, assuming this wouldn't cause disk issues. (Also worth noting none of the workers send with `PERSISTENT_TEXT_PLAIN`)

Comment (dermot-hardy) [reply job-service/src/main/java/com/hpe/caf/services/job/queue/QueueServices.java:143]: Really, what are the implications?  Do we need to raise a Jira for it?

Comment (dgibson-microfocus) [reply job-service/src/main/java/com/hpe/caf/services/job/queue/QueueServices.java:143]: So the persistent flag means messages will be written to disk immediately on arrival to the queue regardless of current memory levels. This means on durable queues any persistent messages will be recovered if the node restarts as opposed to transient messages which are only ever stored in memory (unless near overflow) and are lost if the node goes down.

This is all separate from mirrored queues (which we have in prod) and only really applies to single node recovery. It only comes into play if multiple nodes die before the messages can be replicated. (WHich seems very unlikely)

Comment (dermot-hardy) [reply job-service/src/main/java/com/hpe/caf/services/job/queue/QueueServices.java:143]: Ok.  I assume the acknowledgement is never received until the message has been secured.  If so then it sounds like it's not an issue.

**Pull request merged**